src/formatDate.js

```diff
@@ -50,7 +50,64 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = date.split(fromFormat[3]);
+  const year = {
+    from: {
+      start: false,
+      mid: false,
+      end: false,
+      2: false,
+      4: false,
+    },
+  };
+
+  year.to = { ...year.from };
+
+  const positions = ['start', 'mid', 'end'];
+
+  for (let i = 0; i < positions.length; i++) {
+    if (fromFormat[i].includes('Y')) {
+      year.from[positions[i]] = true;
+      year.from[fromFormat[i].length] = true;
+    }
+
+    if (toFormat[i].includes('Y')) {
+      year.to[positions[i]] = true;
+      year.to[toFormat[i].length] = true;
+    }
+  }
+
+  const index = fromFormat.findIndex((element) => element.includes('Y'));
+  const y = year.from[2]
+    ? (newDate[index] < 30 ? '20' : '19') + newDate[index]
+    : newDate[index].slice(2);
+
+  switch (true) {
+    case (year.from.mid && year.to.end)
+      || (year.from.start && year.to.mid):
```

Comment:
would be better if you created separate variables for conditions. Now it is quite difficult to read